rita_common/src/usage_tracker/mod.rs

```diff
@@ -133,44 +160,107 @@ impl UsageTracker {
             payments: VecDeque::new(),
         };
 
-        match file {
-            Ok(mut file) => {
-                let mut byte_contents = Vec::new();
-                // try compressed
-                match file.read_to_end(&mut byte_contents) {
-                    Ok(_bytes_read) => {
-                        let decoder = ZlibDecoder::new(&byte_contents[..]);
-                        // Extract data from decoder, note this is streaming decoding, so we're
-                        // decompressing the data as we take it out of the zlib compressed object
-                        trace!("attempting to unzip or read bw history");
-                        let deserialized: Result<UsageTracker, SerdeError> =
-                            serde_json::from_reader(decoder);
-
-                        match deserialized {
-                            Ok(value) => value,
-                            Err(e) => {
-                                error!(
-                                    "Failed to deserialize bytes in compressed bw history {:?}",
-                                    e
-                                );
-                                blank_usage_tracker
-                            }
+        if Path::new(&settings::get_rita_common().network.usage_tracker_file).exists()
+            && Path::new(&settings::get_rita_common().network.usage_tracker_file)
+                .extension()
+                .unwrap()
+                .eq("bincode")
+        {
+            let file = File::open(settings::get_rita_common().network.usage_tracker_file);
+            let mut fileopen = match file {
+                Ok(file) => file,
+                Err(e) => {
+                    error!("Failed to open usage tracker file! {:?}", e);
+                    return blank_usage_tracker;
+                }
+            };
+            let mut byte_contents = Vec::new();
+            // try compressed
+            match fileopen.read_to_end(&mut byte_contents) {
```

Comment:
The first two steps here, load file, zlib unzip can be combined into a function that takes file paths. Or in this case since both the if and else condition read the same variable for file path the entire operation can be done once. 

What you might find useful is the idea of doing this with a match statement rather than an if/else tree. It's easier to understand IMO.

Matching on a tuple like this allows you to fold a lot of logic into a singe line.

let file_exists = exits check
let file_is_bincode = bincode check
let grabbed_bytes_and_zlip = function that reads the file and unzlib's the bytes
match (file_exists, file_is_bincode, grabbed_bytes_and_zlip_decompressed) 
{
   (true, true, Ok(bytes)) => bincode deserialization,
  (true, false, Ok(bytes)) => serde deserialization
  ...
}

The great thing about a multi-variable match is that it will tell you all the cases you are not handling and make you consider them, something that's very difficult in a deeply nested if else tree.